packages/synapse-react-client/src/lib/containers/entity/metadata/EntityModal.tsx

```diff
@@ -60,147 +71,167 @@ export const EntityModal: React.FC<EntityModalProps> = ({
   const isLatestVersion =
     isVersionable && (entityBundle.entity as VersionableEntity).isLatestVersion!
 
-  let primaryAction
-  let secondaryActions
-
-  if (!entityBundle) {
-    primaryAction = { skeleton: true }
-    secondaryActions = undefined
-  } else {
-    if (!window.location.href.includes(entityId)) {
-      primaryAction = {
-        skeleton: false,
-        copy: `Open ${entityTypeToFriendlyName(entityBundle.entityType)}`,
-        onClick: () =>
-          window.open(
-            `${getEndpoint(
-              BackendDestinationEnum.PORTAL_ENDPOINT,
-            )}#!Synapse:${entityId}`,
-            '_blank',
-            'noopener',
-          ),
+  const showOpenEntityPageButton =
```

Comment:
Way easier to just create the buttons than to try to replicate this `action` -> `Button` mapping